pytest_httpserver/httpserver.py

```diff
@@ -64,6 +64,22 @@ class NoMethodFoundForMatchingHeaderValueError(Error):
     pass
 
 
+# Note:
+# The following class can be a namedtuple but collections.namedtuple's attributes are not
+# typed and it is better to have a typed object here, it helps the developer using an IDE.
+# Dataclasses have no [0] access.
```

Comment:
why would you want to access request by log[0] and response by log[1]?
imo it is not needed and simple dataclass would suffice.
```
@dataclass
class LogEntry:
     request: Request
     response: Response
```

Reply:
I would like to improve the way we record request-response pairs, and also keep the API compatible. This is why it is not a two-element tuple, but a new object which has request and response attributes.

Currently there's the log attribute of the httpserver object which would store LogEntry objects in the future (this change is not included in this commit, I wanted to commit it separately) and for this, there's the requirement of compatibility as it currently contains a 2-element tuple, so code using [0] and [1] will not break.
I may change the implementation a bit as currently if someone changes the request or response attribute it will be a split-brain, as the [0] and [1] will remain the same. Probably adding two properties for request and response would do the job but I'll see.